open_per_day.rb

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env ruby
+
+require 'optparse'
+require 'csv'
+require 'octokit'
+require_relative 'octokit_utils'
+
+options = {}
+options[:oauth] = ENV['GITHUB_COMMUNITY_TOKEN'] if ENV['GITHUB_COMMUNITY_TOKEN']
+parser = OptionParser.new do |opts|
+  opts.banner = 'Usage: open_and_created.rb [options]'
+
+  opts.on('-n', '--namespace NAME', 'GitHub namespace. Required.') { |v| options[:namespace] = v }
+  opts.on('-r', '--repo-regex REGEX', 'Repository regex') { |v| options[:repo_regex] = v }
+  opts.on('-t', '--oauth-token TOKEN', 'OAuth token. Required.') { |v| options[:oauth] = v }
+  opts.on('-o', '--overview', 'Output overview, summary totals to csv') { options[:display_overview] = true}
+
+  # default filters
+  opts.on('--puppetlabs', 'Select Puppet Labs\' modules') {
+    options[:namespace] = 'puppetlabs'
+    options[:repo_regex] = '^puppetlabs-'
+  }
+  opts.on('--puppetlabs-supported', 'Select only Puppet Labs\' supported modules') {
+    options[:namespace] = 'puppetlabs'
+    options[:repo_regex] = OctokitUtils::SUPPORTED_MODULES_REGEX
+  }
+  opts.on('--community', 'Select community modules') {
+    options[:namespace] = 'puppet-community'
+    options[:repo_regex] = '^puppet-'
+  }
+end
+
+parser.parse!
+
+missing = []
+missing << '-n' if options[:namespace].nil?
+missing << '-t' if options[:oauth].nil?
+if not missing.empty?
+  puts "Missing options: #{missing.join(', ')}"
+  puts parser
+  exit
+end
+
+options[:repo_regex] = '.*' if options[:repo_regex].nil?
+
+util = OctokitUtils.new(options[:oauth])
+repos = util.list_repos(options[:namespace], options)
+
+all_pulls = []
+pr_cache = []
+
+repos.each do |repo|
+
+  #Retrieves all PRs for the repo
+  pr_cache = (util.fetch_async("#{options[:namespace]}/#{repo}", search_with={:state => 'open'}, filter=[]))
+  pr_cache.concat(util.fetch_async("#{options[:namespace]}/#{repo}", search_with={:state => 'closed'}, filter=[]))
+
+  pr_cache.each do |pr|
+    all_pulls.push (pr[:pull])
+  end
+end
+
+puppet_members = {}
+puppet_members = util.puppet_organisation_members(all_pulls)
+
+#Defines the dates required
+end_date = Time.now.to_date
+start_date = end_date - 28
+
+#Currently open per day
+days = []
+(start_date..end_date).each do |day_to_check|
+  puppet_prs = 0
+  community_prs = 0
+  all_pulls.each do |pull|
```

Comment:
This seems to use the same data and queries as daily_open_prs. please merge the two scripts to avoid unnecessary github querying.